document-store/src/main/java/org/hypertrace/core/documentstore/mongo/MongoCollection.java

```diff
@@ -475,6 +466,44 @@ public CloseableIterator<Document> aggregate(
     return convertToDocumentIterator(queryExecutor.aggregate(query));
   }
 
+  @Override
+  public Optional<Document> update(
+      final org.hypertrace.core.documentstore.query.Query query,
+      final java.util.Collection<SubDocumentUpdate> updates)
+      throws IOException {
+    try {
+      final BasicDBObject selections = getSelections(query);
+      final BasicDBObject sorts = getOrders(query);
+      final FindOneAndUpdateOptions options = new FindOneAndUpdateOptions();
```

Comment:
nit: for readability, we can pass the argument `ReturnDocument.BEFORE` or put the comment that it is the default.

Reply:
Sure. Will do.